src/main/java/jarvis/commands/UserList.java

```diff
@@ -0,0 +1,146 @@
+    private static ArrayList<Task> userList= new ArrayList<>(); //list of Tasks
+
+    public UserList(String description) {
+        super(description);
+    }
+
+    public static int getListSize() {
+        return userList.size();
+    }
+
+    public static void insertTask(Task newTask) {
+        userList.add(newTask);
+        DisplayMessages.taskAdded(newTask.getDescription());
+    }
+
+    public static void printList() { //consider converting this function to Jarvis.DisplayMessages
+
+        if (!userList.isEmpty()) {
+            DisplayMessages.horizontalLine();
+            for (int i = 0; i < userList.size(); i++) {
+                String taskIndex = Integer.toString(i+1) + ".";
+                System.out.println(taskIndex.toString() + userList.get(i).getFullTask());
+
+            }
+            DisplayMessages.horizontalLine();
+        } else {
+            DisplayMessages.emptyList();
+        }
+    }
+
+    public static String getStatusIcon(int index){
+        return userList.get(index).getStatusIcon();
+    }
+
+    public static String getTask(int index) {
+        return userList.get(index).getDescription();
+    }
+
+
+    public static void markTask(int index, boolean toPrintMessage) {
+        Task targetTask = userList.get(index);
+        DisplayMessages.horizontalLine();
+        boolean isSuccessfullyMarked = targetTask.markAsDone();
+        if (isSuccessfullyMarked && toPrintMessage) {
+            DisplayMessages.printTaskMarked(targetTask);
+            DisplayMessages.horizontalLine();
+        }
+    }
+
+    public static void unmarkTask(int taskIndex) {
+        Task targetTask = userList.get(taskIndex);
+        DisplayMessages.horizontalLine();
+        boolean isSuccessfullyUnmarked = targetTask.markAsUndone();
+        if (isSuccessfullyUnmarked) {
+            DisplayMessages.printTaskUnmarked(targetTask);
+        }
+    }
+
+    public static void removeTask(int taskIndex) {
+        Task taskRemoved = userList.get(taskIndex);
+        userList.remove(taskIndex);
+        DisplayMessages.taskDeleted(taskRemoved, userList.size());
+    }
+
+    protected static void parseSavedData(String data) {
+        String[] dataArray = data.split(" ");
+        String taskType = dataArray[0];
+        int separatorIndex = Formatter.indexOf(dataArray, "|");
+        String description = "";
+        String date = "";
+        boolean taskIsDone = dataArray[1].equals("YES") ? true : false;
+        if (separatorIndex == -1) {
+            description = Formatter.parseUserInput(dataArray, 2, dataArray.length);
+        } else {
+            //saved Task is either an event or deadline
+            description = Formatter.parseUserInput(dataArray, 2, separatorIndex);
+            date = Formatter.parseUserInput(dataArray, separatorIndex+1, dataArray.length);
+        }
+
+        switch (taskType) {
+        case ("T"):
+            userList.add(new Task(description));
+            break;
+        case("D"):
```

Comment:
Avoid using magic literals such as `"D"`, consider extracting it out into a constant.